design-proposals/feature-lifecycle.md

```diff
@@ -0,0 +1,260 @@
+should progress in its lifecycle.
+
+Eventually we would like to see features being evaluated carefully
+before they are introduced, while they are experimented with and
+proven to be actually in use (and useful) before graduating.
+
+### Goals
+- Define the process a feature needs to pass for graduation.
+- Define the process a feature needs to pass in order to be dropped.
+- Provide policies and rules on how to manage a feature during its
+  lifetime.
+
+### Non Goals
+- Implement enforcement tooling to keep features in sync with
+  the lifecycle rules.
+
+### Definition Of Users
+- Development contributors.
+- Cluster operators.
+
+### User Stories
+- As a Kubevirt contributor, I would like to introduce a new useful
+  feature and follow it to graduation (GA).
+- As a Kubevirt contributor, I would like to discontinue a feature
+  that has not yet reached (formal) graduation.
+- As a Kubevirt contributor, I would like to discontinue a feature
+  that has already graduated.
+- As a Kubevirt cluster operator, I would like to experiment with
+  an undergraduate feature.
+- As a Kubevirt cluster operator, I would like to evaluate an
+  undergraduate feature.
+- As a Kubevirt cluster operator, I would like to keep using a feature
+  that got graduated after I used it during the evaluation period.
+- As a Kubevirt cluster operator, I would like to know that
+  an undergraduate feature is planned to be discontinued.
+- As a Kubevirt cluster operator, I would like to stop using a feature
+  that got discontinued.
+
+### Repos
+This is a cross repo project policy under the
+[kubevirt](https://github.com/kubevirt) organization.
+
+## Proposal Design
+The proposal on how to define a feature lifecycle is influenced by
+processes and policies from the Kubernetes project.
+These sources are scattered around, each focusing on different
+aspects of a feature:
+- [Feature Gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/)
+- [Graduation](https://kubernetes.io/blog/2020/08/21/moving-forward-from-beta/)
+- [Changing the API](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md)
+- [Deprecation](https://kubernetes.io/docs/reference/using-api/deprecation-policy/)
+
+The proposal takes the top-down approach, starting with the high level
+flow that a common feature will traverse through.
+Continuing with actions that need to be taken and timeline suggestions.
+
+Both feature graduation and discontinuation flows are covered.
+Including the implications on users.
+
+Depending on individual topics, follow-up proposals may extend the basic
+points raised in this proposal.
+
+### Feature Stages
+A feature is expected to pass through the following stages:
+1. Enhancement proposal.
+2. Implementation.
+3. Release as Alpha (experimental).
+4. Release as Beta (pre-release for evaluation).
+5. Release as GA (graduation).
+6. Discontinuation.
```

Comment:
How do you define discontinuation?

Maintenance mode - No further development, besides security fixes
Deprecated - Recommended to move off it, still security fixes
…?

Reply:
The details on how to discontinue a feature is describe in detail later in this document.
At this location, a more abstract view of the feature stages is given, I think that adding details about "how" is less useful.